titanic.py

```diff
@@ -93,12 +93,46 @@ def maximum_energy_delta(bqm):
                      for j in bqm.iter_neighbors(i))
                for i in bqm.iter_variables())
 
-def run_demo():
+def mutual_information_bqm(dataset, features):
+    """Build a QUBO that maximizes MI between survival and a subset of features"""
+    variables = ((feature, -mutual_information(prob(dataset[['survived', feature]].values), 1))
+                 for feature in features)
+    interactions = ((f0, f1, -conditional_mutual_information(prob(dataset[['survived', f0, f1]].values), 1, 2))
+                    for f0, f1 in itertools.permutations(features, 2))
+    return dimod.BinaryQuadraticModel(variables, interactions, 0, dimod.BINARY)
+
+def add_combination_penalty(bqm, k, penalty):
+    """Create a new BQM with an addition k-combination penalty"""
```

Comment:
Wording, maybe:
```suggestion
    """Create a new BQM with addition of k-combination penalty"""
```

Reply:
The function only creates a new BQM as a step to updating the original BQM with the k-combination penalty. 
Maybe "Add a k-combination penalty to the given BQM." 
Or even "Add a k-of-n penalty to the given BQM."

Reply:
done